x-pack/platform/plugins/shared/onechat/server/utils/a2a/kibana_agent_executor.ts

```diff
@@ -29,25 +35,32 @@ export class KibanaAgentExecutor implements AgentExecutor {
   ) {}
 
   async execute(requestContext: RequestContext, eventBus: ExecutionEventBus): Promise<void> {
-    const { taskId, userMessage } = requestContext;
+    const { taskId, userMessage, contextId } = requestContext;
 
     try {
-      this.logger.debug(`A2A: Starting task ${taskId}`);
+      this.logger.debug(`A2A: Starting task ${taskId} with contextId ${contextId}`);
 
       // Extract text from message parts
       const userText = userMessage.parts
         .filter((part: Part): part is TextPart => part.kind === 'text')
         .map((part: TextPart) => part.text)
         .join(' ');
 
-      // Execute chat with onechat service
-      const { chat } = this.getInternalServices();
+      // Get services
+      const { chat, conversations } = this.getInternalServices();
+
+      // Prefix conversation id with a2a- to avoid conflicts with existing conversations
+      const a2aConversationId = generateA2AConversationId(contextId);
```

Comment:
Suggestion - add a parameter `autoCreateConversationWithId` which is false by default. move the code to check if conversation exist + create in converse if boolean is true.

Reply:
As discussed in slack added this in https://github.com/elastic/kibana/pull/230994/commits/27117b0ae2c864edb63afbdd05e120b7125121e2

Status quo in chat service:
- if `conversationId` undefined, create a new convo with random id
- if `conversationId` provided: it was assuming that it exists, otherwise throws error

After my change:
- if `conversationId` undefined, create a new convo with random id
-  if `conversationId` provided
  - if  `autoCreateConversationWithId` false (default) , then status quo
  - if `autoCreateConversationWithId` we create new convo with given id
  
  
 let me know id you have any objections to the code etc, if we are fine I will add tests for chat service in next commit as a part of this pr